4.7 Monitoring release notes (#28149)
openshift/openshift-docs: release_notes/ocp-4-7-release-notes.adoc

[id="ocp-4-7-monitoring-prometheus-operator-alertmanagerconfig-crd-"]
==== `AlertmanagerConfig` custom resource definition in Prometheus Operator not supported

Modifying Alertmanager configurations by using the `AlertmanagerConfig` custom resource definition (CRD) in Prometheus Operator is not supported.

For more information, see xref:../monitoring/configuring-the-monitoring-stack.html#support-considerations_configuring-the-monitoring-stack[Support considerations for monitoring].

[id="ocp-4-7-monitoring-alerting-rule-changes"]
==== Alerting rule changes

{product-title} 4.7 includes the following alerting rule changes:

* The `etcdBackendQuotaLowSpace` alert is added. The critical alert provides notification if the database size of an etcd cluster exceeds the defined quota on an etcd instance.
* The `etcdHighFsyncDurations` alert is added. The critical alert provides notification if the 99th percentile fsync durations of an etcd cluster are too high.
* The `etcdExcessiveDatabaseGrowth` alert is added. The warning alert provides notification if there is an observed surge in etcd writes that caused a 50% increase in database size on an etcd instance over a four-hour period.

> * The `NodeClockNotSynchronising` alert is updated ... // Need reason for update. I see that the `expr` changed to now include `and node_timex_maxerror_seconds >= 16`.

Reviewer: cc @paulfantom who modified the alert

Reviewer 2: The reason was to prevent false-positives in environments with chronyd.

> * The `NodeNetworkReceiveErrs` alert is updated ... // Need reason for update. I see that the `expr` changed to `rate(node_network_receive_errs_total[2m]) / rate(node_network_receive_packets_total[2m]) > 0.01`.

Reviewer: The reason behind the change is to make sure that the alert doesn't fire when only a handful number of errors is reported. Rather than using the absolute number of errors, the rule now uses the ratio of errors to total packets.

> * The `NodeNetworkTransmitErrs` alert is updated ... // Need reason for update. I see that the `expr` changed to `rate(node_network_transmit_errs_total[2m]) / rate(node_network_transmit_packets_total[2m]) > 0.01`.

Reviewer: same as above

*

[NOTE]
====
Red Hat does not guarantee backward compatibility for metrics, recording rules, or alerting rules.
====

[id="ocp-4-6-monitoring-namespace-by-pod-and-pod-total-Grafana-dashboards"]
==== Namespace By Pod and Pod Total Grafana dashboards

The Namespace By Pod and Pod Total networking Grafana dashboards are enabled. // What are these dashboards used for?

[id="ocp-4-6-monitoring-prometheus-config-reloader-memory-limits"]
==== `prometheus-config-reloader` memory limit removed for monitoring user defined projects

> The memory limit was removed on the Prometheus Operator `prometheus-config-reloader` container in the `openshift-user-workload-monitoring` namespace for monitoring user defined projects. Memory limits are not set on containers. // More info needed?

Reviewer: The reason behind this change is to avoid the `prometheus-config-reloader` container being killed because it uses more memory than the defined limit.

// TODO: Link to troubleshooting using this dashboard section once it is available

[id="ocp-4-7-monitoring-alerting-rule-changes"]
==== Alerting rule changes

> {product-title} 4.7 includes the following alerting rule changes:

Reviewer 3: 4.7 also added
```
AlertmanagerClusterCrashlooping
AlertmanagerClusterDown
AlertmanagerClusterFailedToSendAlerts
AlertmanagerFailedToSendAlerts

KubeletClientCertificateRenewalErrors
KubeletServerCertificateRenewalErrors

NTODegraded
NTOPodsNotReady

PrometheusOperatorNotReady
PrometheusOperatorRejectedResources
PrometheusOperatorSyncFailed
PrometheusTargetLimitHit
```
removed since 4.7
```etcdHighNumberOfFailedHTTPRequests```

47 rules see from
https://drive.google.com/file/d/1gjXXMnEsEKP2bCD1X3DZ_-W-K9f25Ci-/view?usp=sharing

46 rules see from
https://drive.google.com/file/d/1TXYR7qn10nh84gM-A866R_JdOoDNgD6d/view?usp=sharing

Author: @juzhao @simonpasquier PTAL at the additional new alerts added (see juzhao's previous comment). I also included the removed `etcdHighNumberOfFailedHTTPRequests` alerts (warning and critical).

* The `NodeClockNotSynchronising` alert is updated to prevent false positives in environments that use the chrony time service (chronyd).
* The `NodeNetworkReceiveErrs` alert is updated to ensure that the alert does not fire when only a small number of errors are reported. The rule now uses the ratio of errors to total packets instead of the absolute number of errors.
* The `NodeNetworkTransmitErrs` alert is updated to ensure that the alert does not fire when only a small number of errors are reported. The rule now uses the ratio of errors to total packets instead of the absolute number of errors.

[NOTE]
====
Red Hat does not guarantee backward compatibility for metrics, recording rules, or alerting rules.
====

[id="ocp-4-7-monitoring-stack-dependency-version-updates"]
==== Version updates to monitoring stack components and dependencies

{product-title} 4.7 includes version updates to the following monitoring stack components and dependencies:

* The Prometheus Operator is now on version 0.44.1.
* Thanos is now on version 0.17.2.

> * `kube-prometheus` is now on release 0.7.

Reviewer 4: version of kube-prometheus is not necessary

> * `kube-prometheus` is now on release 0.7.

Reviewer 3: I don't think it is needed, we could remove it

Author: Thank you, @juzhao. @simonpasquier or @s-urbaniak, can you please confirm. This is in the CHANGELOG.

Reviewer: I'd agree with @juzhao, kube-prometheus isn't an operand, it's only configuration.

[id="ocp-4-7-monitoring-prometheus-operator-alertmanagerconfig-crd-"]
==== `AlertmanagerConfig` custom resource definition in Prometheus Operator not supported

> Modifying Alertmanager configurations by using the `AlertmanagerConfig` custom resource definition (CRD) in the Prometheus Operator is not supported.

Reviewer 3: 4.7 also added
```AlertmanagerClusterCrashlooping
AlertmanagerClusterDown
AlertmanagerClusterFailedToSendAlerts
AlertmanagerFailedToSendAlerts

etcdBackendQuotaLowSpace
etcdExcessiveDatabaseGrowth

KubeletClientCertificateRenewalErrors
KubeletServerCertificateRenewalErrors

NTODegraded
NTOPodsNotReady

PrometheusOperatorNotReady
PrometheusOperatorRejectedResources
PrometheusOperatorSyncFailed
PrometheusTargetLimitHit
ThanosSidecarPrometheusDown
ThanosSidecarUnhealthy```

removed since 4.7
```etcdHighNumberOfFailedHTTPRequests```

Reviewer 3: removed since 4.7
```etcdHighNumberOfFailedHTTPRequests```

Reviewer 3: 47 rules see from
https://drive.google.com/file/d/1gjXXMnEsEKP2bCD1X3DZ_-W-K9f25Ci-/view?usp=sharing

46 rules see from
https://drive.google.com/file/d/1TXYR7qn10nh84gM-A866R_JdOoDNgD6d/view?usp=sharing

> For more information, see xref:../monitoring/configuring-the-monitoring-stack.html#support-considerations_configuring-the-monitoring-stack[Support considerations for monitoring].

Author: > The link is not available and get 404 error when access

Thank you, @lihongyan1. I checked the xref in [the preview](https://deploy-preview-28149--osdocs.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-7-release-notes.html#ocp-4-7-monitoring-prometheus-operator-alertmanagerconfig-crd-) and it routes you to the "Support considerations for monitoring." Will double check before merging.

[id="ocp-4-7-monitoring-namespace-by-pod-and-pod-total-Grafana-dashboards"]
==== Namespace By Pod and Pod Total Grafana dashboards

The Namespace By Pod and Pod Total networking Grafana dashboards are now enabled.

[id="ocp-4-7-monitoring-hwmon-hardware-telemetry-bare-metal"]
==== HWMon data collection for hardware telemetry for bare metal clusters

HWMon data collection is enabled for hardware health telemetry such as CPU temperature and fan speeds for bare metal clusters.

[id="ocp-4-7-thanos-query-log-level-config"]
==== Log level configuration field for Thanos Querier

You can now configure the Thanos Querier `logLevel` field for purposes such as debugging.

[id="ocp-4-7-monitoring-prometheus-config-reloader-memory-limits"]
==== Removed `prometheus-config-reloader` memory limit for monitoring user defined projects

> The memory limit was removed on the Prometheus Operator `prometheus-config-reloader` container in the `openshift-user-workload-monitoring` namespace for monitoring user defined projects. This update prevents OOM kill of the `prometheus-config-reloader` container because it used more memory than the defined limit.

Reviewer 4: no prometheus-config-reloader container on prometheus-operator pod, please make it more clear

Reviewer 3: should be `config-reloader` container, the memory limit is removed since 4.7, see the memory limit in a 4.6 cluster
```prometheus-user-workload-0
Container Name: prometheus
resources: map[requests:map[cpu:6m memory:30Mi]]
Container Name: config-reloader
resources: map[limits:map[memory:25Mi] requests:map[cpu:1m memory:10Mi]]```

Author: @lihongyan1 @juzhao @simonpasquier PTAL at the updated wording for the `config-reloader` container memory limit. Thank you.

{product-title} 4.7 includes the following alerting rule changes:

.Alerting rule changes
[%collapsible]
====
* The `AlertmanagerClusterCrashlooping` alert is added. The critical alert provides notification if at least half of the Alertmanager instances in a cluster are crashlooping.
* The `AlertmanagerClusterDown` alert is added. The critical alert provides notification if at least half of the Alertmanager instances in a cluster are down.
* The `AlertmanagerClusterFailedToSendAlerts` alert is added. The critical alert provides notification if all Alertmanager instances in a cluster failed to send notifications.
* The `AlertmanagerFailedToSendAlerts` alert is added. The warning alert provides notification if an Alertmanager instance failed to send notifications.
* The `etcdBackendQuotaLowSpace` alert is added. The critical alert provides notification if the database size of an etcd cluster exceeds the defined quota on an etcd instance.
* The `etcdExcessiveDatabaseGrowth` alert is added. The warning alert provides notification if there is an observed surge in etcd writes that caused a 50% increase in database size on an etcd instance over a four-hour period.

> * The `etcdHighFsyncDurations` alert is added. The critical alert provides notification if the 99th percentile fsync durations of an etcd cluster are too high.

Reviewer 5: Doesn't look like fsync has been marked up in the rest of the repo yet, but... maybe `fsync`?

Reviewer 3: etcdHighFsyncDurations is added to 4.7
```
            "name": "etcdHighFsyncDurations",
            "query": "histogram_quantile(0.99, rate(etcd_disk_wal_fsync_duration_seconds_bucket{job=~\".*etcd.*\"}[5m])) > 0.5",
```

Author: > Doesn't look like fsync has been marked up in the rest of the repo yet, but... maybe `fsync`?

Looks like the etcd docs has `fsync` marked up, too. 👍

* The `NTOPodsNotReady` alert is added. The warning alert provides notification if a specific pod on a node is not ready.
* The `PrometheusOperatorNotReady` alert is added. The warning alert provides notification if the Prometheus Operator in your namespace is not ready.
* The `PrometheusOperatorRejectedResources` alert is added. The warning alert provides notification if specific resources are rejected by the Prometheus Operator.
* The `PrometheusOperatorSyncFailed` alert is added. The warning alert provides notification if the last controller in your namespace failed to reconcile specific objects.
* The `PrometheusTargetLimitHit` alert is added. The warning alert provides notification if Prometheus has dropped targets because some scrape configurations have exceeded the limit of the targets.
* The `ThanosSidecarPrometheusDown` alert is added. The critical alert provides notification that the Thanos sidecar cannot connect to Prometheus.
* The `ThanosSidecarUnhealthy` alert is added. The critical alert provides notification that the Thanos sidecar is unhealthy for a specified amount of time.

> * The `NodeClockNotSynchronising` alert is updated to prevent false positives in environments that use the chrony time service (chronyd).

Reviewer 5: Probably markup `chronyd`. Maybe as "the chrony time service, `chronyd`."

[NOTE]
=====
Red Hat does not guarantee backward compatibility for metrics, recording rules, or alerting rules.
=====
====

[id="ocp-4-7-monitoring-stack-dependency-version-updates"]
==== Version updates to monitoring stack components and dependencies

{product-title} 4.7 includes version updates to the following monitoring stack components and dependencies:

* The Prometheus Operator is now on version 0.44.1.
* Thanos is now on version 0.17.2.
* `kube-prometheus` is now on release 0.7.

[id="ocp-4-7-monitoring-prometheus-operator-alertmanagerconfig-crd-"]
==== AlertmanagerConfig CRD in Prometheus Operator not supported

Modifying Alertmanager configurations by using the `AlertmanagerConfig` custom resource definition (CRD) in the Prometheus Operator is not supported.

For more information, see xref:../monitoring/configuring-the-monitoring-stack.html#support-considerations_configuring-the-monitoring-stack[Support considerations for monitoring].

[id="ocp-4-7-monitoring-namespace-by-pod-and-pod-total-Grafana-dashboards"]
==== Namespace By Pod and Pod Total Grafana dashboards

> The Namespace By Pod and Pod Total networking Grafana dashboards are now enabled.

Reviewer 5: Should these dashboard names be marked up as a `*UI element*`?

Author: Yes, good catch!

* The `etcdBackendQuotaLowSpace` alert is added. The critical alert provides notification if the database size of an etcd cluster exceeds the defined quota on an etcd instance.
* The `etcdExcessiveDatabaseGrowth` alert is added. The warning alert provides notification if there is an observed surge in etcd writes that caused a 50% increase in database size on an etcd instance over a four-hour period.
* The `etcdHighFsyncDurations` alert is added. The critical alert provides notification if the 99th percentile `fsync` durations of an etcd cluster are too high.
* The `KubeletClientCertificateRenewalErrors` alert is added. The warning alert provides notification if Kubelet failed to renew its client certificate.
* The `KubeletServerCertificateRenewalErrors` alert is added. The warning alert provides notification if Kubelet failed to renew its server certificate.
* The `NTODegraded` alert is added. The warning alert provides notification if the Node Tuning Operator is degraded.
* The `NTOPodsNotReady` alert is added. The warning alert provides notification if a specific pod on a node is not ready.

> * The `PrometheusOperatorNotReady` alert is added. The warning alert provides notification if the Prometheus Operator in your namespace is not ready.

Reviewer: ```suggestion
* The `PrometheusOperatorNotReady` alert is added. The warning alert provides notification if a Prometheus Operator instance is not ready.
```

* The `PrometheusOperatorRejectedResources` alert is added. The warning alert provides notification if specific resources are rejected by the Prometheus Operator.

> * The `PrometheusOperatorSyncFailed` alert is added. The warning alert provides notification if the last controller in your namespace failed to reconcile specific objects.

Reviewer: ```suggestion
* The `PrometheusOperatorSyncFailed` alert is added. The warning alert provides notification if a Prometheus Operator's controller failed to reconcile specific objects.
```

// TODO: Link to troubleshooting using this dashboard section once it is available

[id="ocp-4-7-monitoring-namespace-by-pod-and-pod-total-Grafana-dashboards"]
==== New Namespace By Pod and Pod Total monitoring dashboards

> The *Namespace By Pod* and *Pod Total* networking Grafana dashboards are now enabled. You can access the *Namespace By Pod* and *Pod Total* dashboards from the web console by navigating to *Monitoring* -> *Dashboards* and selecting either dashboard.

Author: @dgrisonnet Can you please provide some details on what the newly enabled Namespace By Pod and Pod Total Grafana dashboards are used for? Thanks! (See the "New API Performance monitoring dashboard" section above this note as an example)

// TODO: Link to troubleshooting using this dashboard section once it is available

> [id="ocp-4-7-monitoring-namespace-by-pod-and-pod-total-Grafana-dashboards"]

Reviewer 6: You could update this anchor to reflect the updated title.

> ==== Enable Namespace (Pods) and Pod Kubernetes networking dashboards in Grafana

Reviewer 6: I wondered if this title meant that administrators could enable the dashboards, or if they have been enabled by default in the release. I wondered if "Enabled" would more clearly describe this, but it doesn't resolve the potential ambiguity.

To resolve this, you could update the title to the following:

"Namespace (Pods) and Pod Kubernetes networking dashboards are enabled in Grafana"

> The *Namespace (Pods)* and *Pod* Kubernetes networking dashboards are now enabled in Grafana. You can access the *Namespace (Pods)* and *Pod* dashboards from the web console by navigating to *Monitoring* -> *Dashboards* -> *Grafana UI*.

Reviewer 6: Is this available in the **Administrator** and **Developer** perspectives? 

If it is only available in the **Administrator** perspective, then you could end the sentence with "...in the **Administrator** perspective."

If this is available in both, you could add details about how to access the Grafana UI in each respective perspective, maybe as bullet points.

Author: Admin perspective only.

> These dashboards provide networking metrics, such as:

Reviewer 6: You could change "These" to "The", although I am not sure that this adds anything really.

> * Current rate of bytes received per namespace or per pods

Reviewer 6: You could potentially change "Current rate" to "The current rate...". The same goes for the other items starting with "Rate..." below, each sentence could start with "The".

Reviewer 6: You could change "per pods" to "per pod" in these two lines.

Author: > You could potentially change "Current rate" to "The current rate...". The same goes for the other items starting with "Rate..." below, each sentence could start with "The".

Going to keep without the "The" to match the "API Performance" release note above.
